waveorder/models/phase2D_3D.py

```diff
@@ -0,0 +1,99 @@
+import torch
+import numpy as np
+from waveorder import optics, util
+
+
+def calc_TF(YX_shape, YX_ps, Z_pos_list, lamb_ill, n_media, NA_ill, NA_obj):
+    frr = util.gen_radial_freq(YX_shape, YX_ps)
+
+    ill_pupil = optics.gen_pupil(frr, NA_ill, lamb_ill)
+    det_pupil = optics.gen_pupil(frr, NA_obj, lamb_ill)
+    Hz_stack = optics.gen_Hz_stack(
+        frr, det_pupil, lamb_ill / n_media, torch.tensor(Z_pos_list)
+    )
+
+    ZYX_shape = (len(Z_pos_list),) + YX_shape
+
+    Hu = torch.zeros(ZYX_shape, dtype=torch.complex64)
+    Hp = torch.zeros(ZYX_shape, dtype=torch.complex64)
+    for z in range(len(Z_pos_list)):
+        Hu[z], Hp[z] = optics.WOTF_2D_compute(
+            ill_pupil, det_pupil * Hz_stack[z]
+        )
+
+    return Hu, Hp
+
+
+def visualize_TF(viewer, Hu, Hp, ZYX_scale):
+    # TODO: consider generalizing w/ phase3Dto3D.visualize_TF
+    arrays = [
+        (torch.imag(Hu).numpy(), "Im(H_u)"),
+        (torch.real(Hu).numpy(), "Re(H_u)"),
+        (torch.imag(Hp).numpy(), "Im(H_p)"),
+        (torch.real(Hp).numpy(), "Re(H_p)"),
+    ]
+
+    for array in arrays:
+        lim = 0.5 * np.max(np.abs(array[0]))
+        viewer.add_image(
+            np.fft.ifftshift(array[0], axes=(1,2)),
```

Comment:
Should we do this with tensors?

Reply:
I found that giving a tensor to `viewer.add_image()` fails, so I found I needed to convert with `.numpy()`. Should I always use `.cpu().numpy()`, or is this issue small enough that we can handle it if we need it?

Reply:
I meant doing the FFT in torch and then converting to NumPy, so you don't have to do it many times upfront.

For now `cpu()` doesn't do anything, just to make it future-proof when we do support GPU.

Reply:
Fixed.